docs/architecture.md

```diff
@@ -102,14 +129,15 @@ my_command.add_command(show_command(decorators=lookup_params))
 ```
 
 Some of these commands understand extra arguments.
-They can be attached by passing a list of `click.Options` via the `decorators` argument.
+They can be attached by passing a list of [`pulp_option`][pulpcore.cli.common.generic.pulp_option] factories via the `decorators` argument.
+These factories act as `click.Option` objects, but provides extra functionality specific to this CLI and thus are preferred over the latter.
```

Comment:
This is not correct. You don't pass the factories, but the `Option`s. And the decorator accepts `click.Option`, while `PulpOption` is a subclass thereof.